evap/results/tools.py

```diff
@@ -171,42 +211,91 @@ def _calculate_results_impl(course):
     return sections
 
 
-def calculate_average_grades_and_deviation(course):
-    """Determines the final average grade and deviation for a course."""
-    avg_generic_likert = []
-    avg_contribution_likert = []
-    dev_generic_likert = []
-    dev_contribution_likert = []
-    avg_generic_grade = []
-    avg_contribution_grade = []
-    dev_generic_grade = []
-    dev_contribution_grade = []
+def get_grade_percentages(has_contributors, has_course_grade_questions, has_course_non_grade_questions):
+    contributions_percentage = settings.CONTRIBUTIONS_PERCENTAGE if has_contributors else 0
+    course_grade_questions_percentage = settings.COURSE_GRADE_QUESTIONS_PERCENTAGE if has_course_grade_questions else 0
+    course_non_grade_questions_percentage = 1 - settings.CONTRIBUTIONS_PERCENTAGE - settings.COURSE_GRADE_QUESTIONS_PERCENTAGE if has_course_non_grade_questions else 0
 
-    for __, contributor, __, results, __ in calculate_results(course):
-        average_likert = avg([result.average for result in results if result.question.is_likert_question])
-        deviation_likert = avg([result.deviation for result in results if result.question.is_likert_question])
-        average_grade = avg([result.average for result in results if result.question.is_grade_question])
-        deviation_grade = avg([result.deviation for result in results if result.question.is_grade_question])
+    if contributions_percentage == 0:
+        course_grade_questions_percentage, course_non_grade_questions_percentage = expand_to_100_percent(course_grade_questions_percentage, course_non_grade_questions_percentage)
+    if course_grade_questions_percentage == 0:
+        contributions_percentage, course_non_grade_questions_percentage = expand_to_100_percent(contributions_percentage, course_non_grade_questions_percentage)
+    if course_non_grade_questions_percentage == 0:
+        contributions_percentage, course_grade_questions_percentage = expand_to_100_percent(contributions_percentage, course_grade_questions_percentage)
```

Comment:
inline what where? which other function? i can only guess what exactly you suggest ;)

Reply:
i proposed to inline `get_grade_percentages`.

Reply:
although it calculates three values depending on the three bools or do you mean a more general "see if you can shorten all this to the bare minimum?" because i currently don't see one inline solution. but that might change once i'm actually on it.

Reply:
nvm. i read "one line" instead of "inline".

Reply:
by inlining i meant to just put the code of the function into the call site and remove the function call. the way i see it this part consists of seven rather trivial lines which are not really harder to parse as a two-line function call with three parameters that are calculated in there.

Reply:
good timing :)

Reply:
check